### PR TITLE
Added CountryCodeField to Conformity

### PR DESCRIPTION
**Pull request by ravila-eb**

### CODE REVIEW

**Comment by andrewgodwin** [comment]
I'd like to see this definitely enforce that country codes are either unicode strings or byte strings - I believe at the moment it will allow either on Python 2

**Comment by facundo-eb** [comment]
Look good to me.

File: conformity/fields/__init__.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
     EmailAddress
 )
 from conformity.fields.geo import (  # noqa
+    CountryCodeField,
```
Comment (nickwilliams-eventbrite) [inline conformity/fields/__init__.py]: We can't import this field here. In fact, we can't even put this field in `conformity.fields.geo` (unless we do something special—I'll point out below). That will make `pycountry` a hard dependency. We're trying to make it a soft dependency (hence why it's an `extra` dependency below). So `CountryCodeField` cannot be imported here at all.

File: conformity/fields/geo.py [hunk 2]
```diff
@@ -1,8 +1,21 @@
 from __future__ import absolute_import, unicode_literals
 
 import attr
+import pycountry
```
Comment (nickwilliams-eventbrite) [inline conformity/fields/geo.py]: As mentioned above, this creates a hard dependency on `pycountry`. We need to make this a soft dependency. So we have two options:
1) Put all of these additions in a separate module; or,
2) Do this:

```
try:
    import pycountry

    _countries_a2 = [c.alpha_2 for c in pycountry.countries]

    class CountryCodeField(Constant):
        ... [the rest] ...
except ImportError:
    pass
```

File: setup.py [hunk 3]
```diff
@@ -43,6 +47,7 @@ def readme():
     test_suite='tests',
     extras_require={
         'currency': currency_requires,
+        'geo': geo_requires,
```
Comment (nickwilliams-eventbrite) [inline setup.py]: We might want to name these `country` and `country_requires`, since the rest of the geo fields (latitude and longitude) don't require this.

File: conformity/fields/country.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import six
+
+import pycountry
+from conformity.error import ERROR_CODE_INVALID, Error
+from conformity.fields.basic import Constant
+from conformity.utils import strip_none
```
Comment (nickwilliams-eventbrite) [inline conformity/fields/country.py]: The import grouping and ordering here is not correct. It should be:

```
import pycountry
import six

from conformity.error import ERROR_CODE_INVALID, Error
from conformity.fields.basic import Constant
from conformity.utils import strip_none
```

File: conformity/fields/geo.py [hunk 5]
```diff
@@ -1,7 +1,6 @@
 from __future__ import absolute_import, unicode_literals
 
 import attr
-
```
Comment (nickwilliams-eventbrite) [inline conformity/fields/geo.py]: Please undo this line removal. The `attr` import should be in a separate group from the `conformity` import.

File: tests/test_fields_geo.py [hunk 6]
```diff
@@ -3,10 +3,7 @@
 import unittest
 
 from conformity.error import Error
-from conformity.fields import (
-    Latitude,
-    Longitude,
-)
+from conformity.fields import Latitude, Longitude
```
Comment (nickwilliams-eventbrite) [inline tests/test_fields_geo.py]: Please undo this change. Multi-imports should be multi-line in parens.
